### PR TITLE
Input capture

### PR DESCRIPTION
**Pull request by FranciscoJT97**
#24 
Definition of done:

- [x] Initialize the input capture module in period measurement mode. 
- [x] Implement a formula to convert the period value to a frequency value.

QA:
- [x] Behaves as expected. 

Connect PWM output (P45) to input capture (P512) and watch in the console of the debugger expressions: `frequency` and `rpm` .

`frequency` should be equal to 1000 (1KHz).
`rpm` should be equal to 15000.

### CODE REVIEW

File: Code/Integrator_Project/src/InputCapture.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ */
+
+#ifndef INPUT_CAPTURE_H
+#define INPUT_CAPTURE_H
+
+#include "hal_data.h"
+
+input_capture_info_t input_capture_info;
+uint32_t rpm;
+void InputCapture(void);
```
Comment (IrvinRamiA) [inline Code/Integrator_Project/src/InputCapture.h]: AWS

File: Code/Integrator_Project/src/hal_entry.c [hunk 2]
```diff
@@ -26,6 +26,8 @@ void hal_entry(void)
     OpenTimer();
     StartTimer();
 
+    InputCapture();
```
Comment (IrvinRamiA) [inline Code/Integrator_Project/src/hal_entry.c]:
```suggestion
    InitializeInputCapture();
```

**Review by JuanLNH — APPROVED** [review]
LGTM!

**Comment by IrvinRamiA** [comment]
LGTM.

**Review by IrvinRamiA — APPROVED** [review]

**Pull request merged**